src/doc/trpl/method-syntax.md

```diff
@@ -86,3 +192,77 @@ This *static method* builds a new `Circle` for us. Note that static methods
 are called with the `Struct::method()` syntax, rather than the `ref.method()`
 syntax.
 
+We can combine a simple version of `new` with our builder pattern to get that
+final behavior we wanted:
+
+```
+struct Circle {
+    x: f64,
+    y: f64,
+    radius: f64,
+}
+
+impl Circle {
+    fn area(&self) -> f64 {
+        std::f64::consts::PI * (self.radius * self.radius)
+    }
+}
+
+struct CircleBuilder {
+    coordinate: f64,
+    radius: f64,
+}
+
+impl CircleBuilder {
+    fn new() -> CircleBuilder {
+        CircleBuilder { coordinate: 0.0, radius: 0.0, }
+    }
+
+    fn coordinate(&mut self, coordinate: f64) -> &mut CircleBuilder {
+	self.coordinate = coordinate;
+
+	self
+    }
+
+    fn radius(&mut self, radius: f64) -> &mut CircleBuilder {
+	self.radius = radius;
+
+	self
+    }
+
+    fn finalize(self) -> Circle {
+        Circle { x: self.coordinate, y: self.coordinate, radius: self.radius }
+    }
+}
+
+fn main() {
+    let mut cb = CircleBuilder::new();
+
+    cb.coordinate(10.0)
+      .radius(5.0);
+
+    let c = cb.finalize();
```

Comment:
Consuming builders are typically done via an in-place chain, so perhaps this could be formatted as:

```rust
let c = CircleBuilder::new()
                      .coordinate(10.0)
                      .radius(5.0)
                      .finalize();
println!("area: {}", c.area());
```